chapters/memory.txt

```diff
@@ -2772,9 +2772,10 @@ the last time it was invalidated.
 [NOTE]
 .Note
 ====
-Mapping non-coherent memory does not implicitly invalidate the mapped
-memory, and device writes that have not been invalidated must: be made
-visible before the host reads or overwrites them.
+Mapping non-coherent memory does not implicitly invalidate that memory.
+Device writes to non-coherent memory need to be made visible to the host
+before any further host access to avoid a data race, irrespective of when
+that memory was mapped to the host.
```

Comment:
- further -> subsequent?
- To avoid a data race, device writes ... ?
- need to be **explicitly** made visible  \ (or: "the App needs to make visible")
- **a** non-coherent memory (I think it is countable noun, as it refers to the one specific object or range, i.e. the parameters of `vkMapMemory`)

- , irrespective of when that memory was mapped to the host
->
always need to be made visible ...
(it is irrelevant whether the memory was mapped after the device write).

Reply:
On second thoughts, I'm just going to suggest we just reduce the note to the first sentence. The rest is just niggle-fuel :P

Reply:
Fine with me

Reply:
Yes, this feels better.

Reply:
> a non-coherent memory (I think it is countable noun, as it refers to the one specific object or range, i.e. the parameters of vkMapMemory)

Oh on this, since I didn't technically resolve it - the usage is correct as-is, it's used a mass noun - it's not a single VkMemory object that gets mapped, rather some range of a larger whole.